new/rfcXXX-policyfile-endpoint.md

```diff
@@ -0,0 +1,558 @@
+
+##### `POST /policy_groups/:policy_group_name/policies/:policy_name`
+
+Sets the current active revision of `:policy_name` for the `:policy_group_name`
+to the value given in the POST body. The revision must already exist on the
+server. For example, to set the active revision of the 'appserver' policy in
+the 'qa' policy group, you would POST to
+`/policy_groups/qa/policies/appserver`, with a POST body like the following:
+
+```json
+{
+  "revision_id": "bfec256f76a52ff707ca72e71b464a41c410b229"
+}
+```
+
+#### `/policy_groups/:policy_group_name/nodes`
+
+##### `GET /policy_groups/:policy_group_name/nodes`
+
+Returns a list of nodes that belong to the policy group. Can be filtered for a
+specific `:policy_name` with the `policy_name` query parameter, using a URL of
+the form: `GET /policy_groups/:policy_group_name/nodes?policy_name=:policy_name`
+
+#### `/policies/:policy_name`
+
+TODO: This is where AuthZ for policies goes, does it do anything else?
+
+#### `/policies/:policy_name/revisions/`
+
+##### `POST /policies/:policy_name/revisions/`
+
+Create a new Policyfile lock document. The POST body MUST be a valid Policyfile
+lock document, as described above. The document can subsequently be retrieved
+at the relative URL `/policies/:policy_name/revisions/:revision_id`
+
+Policyfile lock revisions cannot be updated. If the server already has a
+Policyfile lock document with the same `revision_id` as given in the body of
+the request, the server MUST return 409 response.
+
+##### `GET /policies/:policy_name/revisions/`
+
+Returns a list of the available revisions for the given policy name
+
+#### `/policies/:policy_name/revisions/:revision_id`
+
+##### `GET /policies/:policy_name/revisions/:revision_id`
+
+Returns the policyfile lock document with the given name and revision ID.
+
+##### `DELETE /policies/:policy_name/revisions/:revision_id`
+
+Deletes the policyfile lock document with the given name and revision ID.
+
+#### `/policies/:policy_name/revisions/:revision_id/policy_groups`
+
+##### `GET /policies/:policy_name/revisions/:revision_id/policy_groups`
+
+Returns the list of policy groups that are associated with the given policy
+name at the given revision ID.
+
+### Promotion API (Optional)
+
+In addition to the basic REST interface described above, the server may also
+provide a promotion API. The promotion API facilitates defining the deployment
+cycle for policies (i.e., the order in which policies move through policy
+groups) and allows policies to be promoted to subsequent stages of the
+deployment cycle with a single API call.
+
+#### `/policy_groups/:policy_group_name`
+
+#### `GET /policy_groups/:policy_group_name`
+
+Returns the policy group configuration for `:policy_group_name`. The document
+will contain the name of the next policy group in the deployment cycle. Exact
+format is TBD, but it could look like:
+
+```json
+{
+  "name": "development",
+  "next_group_name": "qa"
```

Comment:
I could see a case for it. For now I'm going to leave it as is and I'll make updates when I get the chance to create this API, since I think I need to start implementing to get a better feel for how these details should work.